Character.cpp

```diff
@@ -98,20 +98,32 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+
+        if( getHP() < *initialHitPoints)
+            hitPoints = *initialHitPoints;
+
+        hitPoints *= 1.1f;
+        armor *= 1.1f;
+        attackDamage *= 1.1f;
+        
+        *initialHitPoints = hitPoints;
+        *initialArmorLevel = armor;
+        *initialAttackDamage = attackDamage;
+
```

Comment:
- You aren't checking if the armor or attackDamage are lower than their initial value.
- you're doing the same thing to 3 pairs of variables
- refactor this into a free function that takes a pair of `int&`
- call that function with the 3 pairs of variables
```
            a) your stats are restored to their initial value if they are lower than it.
            b) your stats are boosted 10%
            c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
```
- [ ] revise